### ISSUE TITLE
Discourse settings being overridden

### ISSUE THREAD

**Issue by tdudgeon** [bitnami/bitnami-docker-discourse#173]
**Description**

I've trying to deploy Discourse with additional configuration as described [here](https://github.com/bitnami/bitnami-docker-discourse#configuration-files).
I'm mounting in a site_settings.yml file with my settings onto the container, and it is being picked up.
However something is overriding these settings (I can see this in the UI if I revert the current setting and it gets set back the the one in my config file).
For some of these I can see that they are being overridden by environment variables so I can define them that way. But for others such as the site description I can find no environment variable that defines this, but it is getting reset to "Powered by Bitnami".
Where are these settings being overridden?

**Steps to reproduce the issue:**

1. create a custom `site_settings.yml` modifying the `required.site_description` and `required.short_site_description` properties
2. mount that into the container at `/opt/bitnami/discourse/mounted-conf`
3. inspect the settings in the UI. The `short_site_description` shows what was in the `site_settings.yml` file but the `site_description` has been overridden to 'Powered by Bitnami' but can be reset to the configured value.

**Describe the results you received:**

Some settings are overriden. 

**Describe the results you expected:**

The values in the `site_settings.yml` would be used unless they are overridden by environment variables.

**Version**

Based on image `bitnami/discourse:2-debian-10`

- Output of `docker version`:

```
Client:
 Version:           19.03.8
 API version:       1.40
 Go version:        go1.13.8
 Git commit:        afacb8b7f0
 Built:             Fri Dec 18 12:15:19 2020
 OS/Arch:           linux/amd64
 Experimental:      false

Server:
 Engine:
  Version:          19.03.8
  API version:      1.40 (minimum version 1.12)
  Go version:       go1.13.8
  Git commit:       afacb8b7f0
  Built:            Fri Dec  4 23:02:49 2020
  OS/Arch:          linux/amd64
  Experimental:     false
 containerd:
  Version:          1.3.3-0ubuntu2.2
  GitCommit:        
 runc:
  Version:          spec: 1.0.1-dev
  GitCommit:        
 docker-init:
  Version:          0.18.0
  GitCommit:        

```

- Output of `docker info`:

```
Client:
 Debug Mode: false

Server:
 Containers: 8
  Running: 4
  Paused: 0
  Stopped: 4
 Images: 1152
 Server Version: 19.03.8
 Storage Driver: overlay2
  Backing Filesystem: <unknown>
  Supports d_type: true
  Native Overlay Diff: true
 Logging Driver: json-file
 Cgroup Driver: cgroupfs
 Plugins:
  Volume: local
  Network: bridge host ipvlan macvlan null overlay
  Log: awslogs fluentd gcplogs gelf journald json-file local logentries splunk syslog
 Swarm: inactive
 Runtimes: runc
 Default Runtime: runc
 Init Binary: docker-init
 containerd version: 
 runc version: 
 init version: 
 Security Options:
  apparmor
  seccomp
   Profile: default
 Kernel Version: 5.8.0-050800-generic
 Operating System: Ubuntu 20.04.1 LTS
 OSType: linux
 Architecture: x86_64
 CPUs: 24
 Total Memory: 62.82GiB
 Name: lithium
 ID: O7T6:UO3H:LPC6:BS6S:VWPN:53SK:QHMA:KUZW:KHKX:VXX7:ASYH:EJBP
 Docker Root Dir: /var/lib/docker
 Debug Mode: false
 Username: tdudgeon
 Registry: https://index.docker.io/v1/
 Labels:
 Experimental: false
 Insecure Registries:
  127.0.0.0/8
 Live Restore Enabled: false


```

- Output of `docker-compose version` (if applicable):

```
docker-compose version 1.25.0, build unknown
docker-py version: 4.1.0
CPython version: 3.8.5
OpenSSL version: OpenSSL 1.1.1f  31 Mar 2020
```

**Additional environment details (AWS, VirtualBox, Docker for MAC, physical, etc.):**

Running using a docker-compose file based on the one in this repo.

**Comment by FraPazGal** [comment]
Hello @tdudgeon,

Thanks for pointing this out! We have been able to reproduce the issue and an internal task has been created for the team to work on this. As this is a low-impact bug, we are not able to give you an ETA. We will update this ticket as soon as there are any news.

**Comment by tdudgeon** [comment]
@FraPazGal Thanks for responding to this. Much appreciated.
I feel that this is medium-impact issue, as we should treat our containers as cattle not pets, so MUST allow everything to be configured at deployment time.
Can you consider re-labelling to a higher priority?
Also, the docs need updating, if/when addressed.

**Comment by FraPazGal** [comment]
Hello @tdudgeon,

Thank you for your feedback! We will re-evaluate its priority and proceed accordingly.

**Comment by tdudgeon** [comment]
Any update on this?
Is there anything we can do to grease the wheels?

**Comment by migruiz4** [comment]
Hi @tdudgeon,

I'm sorry for the late response. 

The Discourse image will receive very soon a major update replacing the container logic with shell scripts as we did for the other images in our catalog.

The new logic should consider your use case and fix this issue.

I will let you know as soon as it is available.

**Comment by marcosbc** [comment]
Hi, we just released a new container version, [2.7.0-debian-10-r4](https://github.com/bitnami/bitnami-docker-discourse#270-debian-10-r4). This version uses initialization scripts located in the [`rootfs/` directory](https://github.com/bitnami/bitnami-docker-discourse/blob/master/2/debian-10/rootfs/).

As a major refactor of the existing logic we are not overriding any settings in Discourse anymore, so feel free to check it out and let us know if this is solved.
